lib/galaxy/datatypes/data.py

```diff
@@ -1062,6 +1064,8 @@ def get_file_peek(file_name, is_multi_byte=False, WIDTH=256, LINE_COUNT=5, skipc
                     skip_line = True
                     break
             if not skip_line:
+                if line.endswith('\n'):
+                    line = line[:-1]
                 lines.append(line)
                 count += 1
-    return '\n'.join(lines)
+    return '\n'.join(lines) + '\n'
```

Comment:
This change makes the `test_get_file_peek` unit test fail. Reverting it may break some other tests, but I'd prefer not to add newlines that may not be at the end of the original file.

Reply:
I was thinking since they are just used in HTML display it wasn't a big deal, but you're right I guess - we shouldn't be adding extra newlines. 🤔